src/cmd/linuxkit/pkglib/build.go

```diff
@@ -191,3 +199,62 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 
 	return nil
 }
+
+type buildCtx struct {
+	sources []string
+}
+
+// Copy iterates over the sources, tars up the content after rewriting the paths.
+// It assumes that sources is sane, ie is well formed and the first part is an absolute path
+// and that it exists. NewFromCLI() ensures that.
+func (c *buildCtx) Copy(w io.WriteCloser) error {
+	tw := tar.NewWriter(w)
+	defer func() {
+		tw.Close()
+		w.Close()
+	}()
+
+	for _, source := range c.sources {
+		tmp := strings.Split(source, ":")
+		src := tmp[0]
+		dst := tmp[1]
+		log.Debugf("Adding to build context: %s -> %s", src, dst)
+
+		f := func(p string, i os.FileInfo, err error) error {
+			if err != nil {
+				return fmt.Errorf("ctx: Walk error on %s: %v", p, err)
+			}
+
+			h, err := tar.FileInfoHeader(i, "")
+			if err != nil {
+				return fmt.Errorf("ctx: Converting FileInfo for %s: %v", p, err)
+			}
+			h.Name = path.Join(dst, strings.TrimPrefix(p, src))
```

Comment:
`p` at this point has, I think, been [`Clean`ed](https://golang.org/pkg/path/filepath/#Clean) relative to the original `src`.

If `src` was unclean, perhaps having `//` or "/../" in it or similar then `TrimPrefix` won't DTRT. It looks like [`filepath.Rel`](https://golang.org/pkg/path/filepath/#Rel) might do exactly what you want. It might be sufficient to `filepath.Clean` both `src` and `dst` yourself, but I think `filepath.Rel` is probably better.